src/control/server/instance_storage_rpc_test.go

```diff
@@ -18,13 +18,154 @@ import (
 	"github.com/daos-stack/daos/src/control/common/test"
 	"github.com/daos-stack/daos/src/control/lib/ranklist"
 	"github.com/daos-stack/daos/src/control/logging"
+	"github.com/daos-stack/daos/src/control/provider/system"
 	"github.com/daos-stack/daos/src/control/server/config"
 	"github.com/daos-stack/daos/src/control/server/engine"
 	"github.com/daos-stack/daos/src/control/server/storage"
 	"github.com/daos-stack/daos/src/control/server/storage/bdev"
 	"github.com/daos-stack/daos/src/control/server/storage/scm"
 )
 
+func TestIOEngineInstance_populateCtrlrHealth(t *testing.T) {
+	mockLspciOut := `01:00.0 Non-Volatile memory controller: Intel Corporation PCIe Data Center SSD (rev 01) (prog-if 02 [NVM Express])
```

Comment:
I think I get what Mike is saying here. I don't think the server package should need to know about how to parse lspci output.

Perhaps in the lib/hardware library there could be some definitions like:
```go
type DeviceLinkStats struct {
        Speed uint64 // maybe specify units?
        Capabilities []string
        // etc... any other useful fields
}

type DeviceLinkStatsProvider interface {
        GetDeviceLinkStats(device string) (*DeviceLinkStats, error)
}
```

And then create an lspci provider that implements this one function. It keeps all the lspci output string parsing down at that low level. Then at all the upper levels, we can interpret and format them as we please. Also if we need to use some other data source later it could be easily changed. It would also be easier to mock.

Reply:
I didn't want to go into that detail within this PR, i.e. parsing each key/val within all of the PCIe config space link parameters. Providing the string value for each of the Lnk* lspci output lines in a health stat seemed like an efficient way of meeting the requirements of the single customer for something that's not particular high priority. To me this feels like over engineering but I will proceed with the consensus.

Reply:
An intermediate approach could be to have the link stats fields be strings, but still create the lib/hardware framework around the implementation. Give the fields the same names as the protobuf fields for easy conversion. That would create a framework that allows for future refinement if we ever need to customize display or otherwise use these fields at a higher level.

My main concern is having the server package parsing stuff from lspci. If it ever becomes a problem it will be harder to understand/replace if assumptions about using lspci are sprinkled around the code and unit tests.